### ISSUE TITLE
does not check for unescaped parenthesēs in certain contexts

### ISSUE THREAD

**Issue by mirabilos** [koalaman/shellcheck#2906]
- [x] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/sh
line=x
  if [ "${line##*AC_CONFIG_AUX_DIR}" != "$line" ]; then
    dirprefix="${line##*([}"
    dirprefix="${dirprefix%%])*}"
    mkdir -p "${dirprefix}"
  fi
```

[source](https://github.com/sabotage-linux/gettext-tiny/blob/master/src/autopoint.in)

#### Here's what shellcheck currently says:

No issues detected!

#### Here's what I wanted or expected to see:

A warning about failure to quote the two parenthesēs with a backslash each:

```
    dirprefix="${line##*\([}"
    dirprefix="${dirprefix%%]\)*}"
```

This is **mandatory** because POSIX allows implementations to extend pattern matching in these situations (C.2.14.1 in the current Issue 8 draft specifically mentions the Korn shell extglob extension of `*(…)` in pattern matching).

Unfortunately, both GNU bash and dash allow this (and so mksh’s lksh binary on Debian when in sh mode does), so this mistake is very widespread. It would be good for shellcheck to flag this as **error.**

Thanks @lanodan for noting.
